state/runtime/tracer/structtracer/tracer.go

```diff
@@ -172,30 +173,36 @@ func (t *StructTracer) captureMemory(
 	}
 }
 
+func copyUint256Array(original []uint256.Int) []uint256.Int {
+	// Create a new array with the same length as the original
+	copied := make([]uint256.Int, len(original))
+
+	// Copy each element individually
+	for i := range original {
+		copied[i].Set(&original[i])
+	}
+
+	return copied
+}
+
 func (t *StructTracer) captureStack(
-	stack []*big.Int,
+	stack []uint256.Int,
 	sp int,
 	opCode int,
 ) {
 	if !t.Config.EnableStack {
 		return
 	}
 
-	currentStack := make([]*big.Int, sp)
-
-	for i, v := range stack[:sp] {
-		currentStack[i] = new(big.Int).Set(v)
-	}
-
-	t.currentStack[len(t.currentStack)-1] = currentStack
+	t.currentStack[len(t.currentStack)-1] = copyUint256Array(stack)
```

Comment:
Shouldn't we preserve only the items within the `[0:sp]` range?

Reply:
Acknowledged, but that was not my point. Prior to these changes the `sp` parameter was used to copy items of the stack only up to the `sp` index. So I'm wondering is this change expected and if it is, do we even need the `sp` parameter (it is unused).

Reply:
We have agreed to remove the `sp` parameter through a separate PR. Resolving the comment.